src/plugins/data_source/common/data_sources/types.ts

```diff
@@ -7,6 +7,20 @@ import { SavedObjectAttributes } from 'src/core/types';
 
 export interface DataSourceAttributes extends SavedObjectAttributes {
   title: string;
+  description?: string;
   endpoint: string;
   noAuth: boolean;
+  credentials?: {
+    type: CredentialsType;
+    credentialsContent: UsernamePasswordTypedContent;
```

Comment:
Should you provide ways to get the mapping between type & credentialsContent type?

Reply:
can you elaborate the mapping?